Term Entry Java Date.from() (#2929)
Codecademy/docs: content/java/concepts/date/terms/from/from.md

---
Title: '.from()' 

Description: 'Returns an instance of a Date'

Subjects:
  - 'Data Science'
  - 'Computer Science'
  - 'Discrete Math'

Tags: 
  - 'Date'
  - 'Time'
  - 'Methods'

CatalogContent: 
  - 'learn-java'
  - 'paths/computer-science'
---

> The **'Date.from'** method returns an instance of a Date that is retrieved from an Instant object.  An Instant is more precise than a Date because it reports past milliseconds to nanoseconds.

Reviewer: ```suggestion
The **`Date.from()`** method returns an `Instant` object of a given `Date`. An `Instant` provides more accuracy than a `Date` because it reports nanoseconds (while `Dates` are limited to milliseconds).
```

## Syntax

```pseudo

> date.from(inst)

Reviewer: ```suggestion
someDate.from(inst)
```

```

> The `Date.from()` method is used in combination with an `Instant` object:

Reviewer: ```suggestion
The `Date.from()` method is used in combination with an `Instant` object `inst`.
```

In Java `from` is not a keyword.  The `from` method can be used with the [Date](https://www.codecademy.com/resources/docs/java/date) class to retrieve the current day, date, time, timezone and year.  

Date today = new Date();
Instant inst = Instant.now();

> System.out.println(today.from(inst));

Reviewer: this should be a code block

> ## Example

Reviewer: add an empty line after the header

> The following example uses the words "apples" and "oranges" to avoid syntax confusion: 

Reviewer: ```suggestion
The following example uses the words `"apples"` and `"oranges"` to avoid syntax confusion: 
```

```java
import java.time.Instant;
import java.util.Date;

class JavaDateDemo {
    public static void main(String args[])
    {
        // This line creates a Date object called "apples".
        Date apples = new Date();

        // This line creates an Instant object called "oranges".
        Instant oranges = Instant.now();

        // This line displays the instant using the .from() method.
        System.out.println(
                "Present day, date, time, timezone and year: "
                        + apples.from(oranges));
    }
}
```

> The output will be the exact day, date, time, timezone and year of the moment the query is submitted.  
> 

Reviewer: ```suggestion

The output will be the exact day, date, time, timezone and year of the moment the query is submitted:
```
